utils/trivia.py

```diff
@@ -57,3 +60,42 @@ def count(self) -> int:
     #  messages
     def __len__(self) -> int:
         return self.count
+
+
+# Trivia game object
+# The purpose of this object is to hold trivia game data and
+# maintain game state.
+class Trivia:
+
+    # Object for holding the list of questions
+    qlist: QuestionList
+
+    # The inProgress variable is used to keep track of
+    # whether a trivia game is in progress or not.
+    inProgress: bool
+
+    def __init__(self):
+        self.qlist = QuestionList([QuestionAnswer("Question 1", "Answer 1")])
+        self.inProgress = False
+
+    # This function is responsible for starting the trivia game.
+    def start(self) -> int:
```

Comment:
Instead of having this function return an int, I was wondering whether it might be more beneficial to have it return the trivia object itself. Then you could access the object and fields like the current question being asked, so that the commands could pass in useful arguments. (For example, the !answer command in the other file could pass in the current question being asked to the check() function in order to compare the answer).

Reply:
The reason the function returns an int is to indicate the success or failure of the operation. This is called a return code. What you are describing is somewhat similar to a factory function and is not necessary in this case. There is no reason for the Trivia object itself to change, only the state of the trivia object. The only thing trivia member functions should be doing is modifying the state of the object. Any information the object holds should be accessed through the accessor functions.

If you were to return a new trivia object then you would have to store that object somewhere so you could do things like see what the current question is. The most sensible place to store it would be in the bot which would mean you would need to replace the existing trivia object with the new one. Doing so would incur a lot of extra overhead from allocating and deallocating objects.